### ISSUE TITLE
Unexpected panic: arithmetic operation overflowed

### ISSUE THREAD

**Issue by kubkon** [rust-lang/rust#23084]
Encountered while compiling https://github.com/kubkon/rust-GSL/commit/fb229ed072079a0a927bdc47491c7fcdfb72b961 on Yosemite. Travis reported the same error.

Rustc version:
```
rustc 1.0.0-nightly (3b3bb0e68 2015-03-04) (built 2015-03-04)
```

Backtrace:
```
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'arithmetic operation overflowed', /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libsyntax/codemap.rs:73

stack backtrace:
   1:        0x110837b52 - sys::backtrace::write::h6b0889bc971c1ad3IDA
   2:        0x110866844 - panicking::on_panic::h0fcd1d74630df38dKsJ
   3:        0x110784c67 - rt::unwind::begin_unwind_inner::h56b969a14fc61916gbJ
   4:        0x1107854be - rt::unwind::begin_unwind_fmt::h35c214d65a920712R9I
   5:        0x11086631d - rust_begin_unwind
   6:        0x1108b6ba5 - panicking::panic_fmt::h33ae2b9049f12718IZs
   7:        0x1108aea37 - panicking::panic::h972558dc77800fbdUXs
   8:        0x10fec6d3f - codemap::BytePos.Sub::sub::h3124d8158f2e0f20fWB
   9:        0x10d706270 - middle::astencode::DecodeContext<'a, 'b, 'tcx>::tr_span::hf5ab3413e1878576FCa
  10:        0x10d7171ab - fold::noop_fold_expr::h6174175571178385447
  11:        0x10d716ed5 - ptr::P<T>::map::h12929723692291469568
  12:        0x10d71aa35 - fold::noop_fold_expr::closure.66290
  13:        0x10d717454 - fold::noop_fold_expr::h6174175571178385447
  14:        0x10d716ed5 - ptr::P<T>::map::h12929723692291469568
  15:        0x10d71aa35 - fold::noop_fold_expr::closure.66290
  16:        0x10d717650 - fold::noop_fold_expr::h6174175571178385447
  17:        0x10d716ed5 - ptr::P<T>::map::h12929723692291469568
  18:        0x10d71a9d2 - fold::noop_fold_expr::closure.66285
  19:        0x10d71f20d - fold::noop_fold_local::closure.66516
  20:        0x10d71ec08 - fold::noop_fold_decl::closure.66513
  21:        0x10d71cce1 - fold::Folder::fold_stmt::closure.66443
  22:        0x10d71c49d - fold::noop_fold_block::closure.66394
  23:        0x10d71c0d1 - iter::FlatMap<I, U, F>.Iterator::next::h9486311241923537267
  24:        0x10d71b89f - vec::Vec<T>.FromIterator<T>::from_iter::h6767650569938704022
  25:        0x10d71b1b4 - fold::noop_fold_block::closure.66303
  26:        0x10d71ac3a - fold::noop_fold_block::h4538952730048595662
  27:        0x10d71839b - fold::noop_fold_expr::h6174175571178385447
  28:        0x10d716ed5 - ptr::P<T>::map::h12929723692291469568
  29:        0x10d71a9d2 - fold::noop_fold_expr::closure.66285
  30:        0x10d71b240 - fold::noop_fold_block::closure.66303
  31:        0x10d71ac3a - fold::noop_fold_block::h4538952730048595662
  32:        0x10d71839b - fold::noop_fold_expr::h6174175571178385447
  33:        0x10d716ed5 - ptr::P<T>::map::h12929723692291469568
  34:        0x10d71d24c - fold::Folder::fold_stmt::closure.66443
  35:        0x10d71c49d - fold::noop_fold_block::closure.66394
  36:        0x10d71c0d1 - iter::FlatMap<I, U, F>.Iterator::next::h9486311241923537267
  37:        0x10d71b89f - vec::Vec<T>.FromIterator<T>::from_iter::h6767650569938704022
  38:        0x10d71b1b4 - fold::noop_fold_block::closure.66303
  39:        0x10d71ac3a - fold::noop_fold_block::h4538952730048595662
  40:        0x10d7179d4 - fold::noop_fold_expr::h6174175571178385447
  41:        0x10d716ed5 - ptr::P<T>::map::h12929723692291469568
  42:        0x10d71a9d2 - fold::noop_fold_expr::closure.66285
  43:        0x10d71b240 - fold::noop_fold_block::closure.66303
  44:        0x10d71ac3a - fold::noop_fold_block::h4538952730048595662
  45:        0x10d73bf5b - fold::noop_fold_method::closure.66897
  46:        0x10d73b736 - fold::noop_fold_method::h14362806825288189584
  47:        0x10d70906e - middle::astencode::decode_inlined_item::h79c9a5b47b57a210Pqa
  48:        0x10d2c794c - trans::inline::instantiate_inline::closure.39583
  49:        0x10db3686f - metadata::decoder::maybe_get_item_ast::h368b0a5a4a94a48eMBi
  50:        0x10d91cac8 - metadata::csearch::maybe_get_item_ast::hc09013c09b0dd4bco2k
  51:        0x10d2c609c - trans::inline::instantiate_inline::h4cdedfa8fc78ffd9Pfd
  52:        0x10d3183d6 - trans::callee::trans_fn_ref_with_substs::heb22ab8904f67bf64kg
  53:        0x10d316b0e - trans::callee::trans_fn_ref::hb2be9d8aec49d9c4E9f
  54:        0x10d32ea7c - trans::meth::trans_method_callee::h92a644d0a2dc2a56N6x
  55:        0x10d32cdbb - trans::callee::trans_call_inner::h17268543188350918970
  56:        0x10d33776a - trans::expr::trans_rvalue_dps_unadjusted::hca2c07ae951484c14zi
  57:        0x10d2e5f59 - trans::expr::trans_into::h4a29f680db4b3526znh
  58:        0x10d2e4fa0 - trans::controlflow::trans_stmt_semi::h318c4f3f34b9c533o4d
  59:        0x10d2e6ace - trans::controlflow::trans_block::h26a308528bb95051b5d
  60:        0x10d3c5da9 - trans::base::trans_closure::h1e78b4450eb13a03Kkt
  61:        0x10d2ce645 - trans::base::trans_fn::h31c9353b85ff9dd9Dvt
  62:        0x10d2ca0bd - trans::base::trans_item::h8ce277cc13b040a3vTt
  63:        0x10d3ceecb - trans::base::trans_crate::hb69c466e59630a14GPu
  64:        0x10ccfbec7 - driver::phase_4_translate_to_llvm::he4e045cdc3897b4crNa
  65:        0x10ccd77f2 - driver::compile_input::h25d96a14ec26932cIba
  66:        0x10cda972e - run_compiler::h15c3ea085a111a6fH5b
  67:        0x10cda6ac7 - thunk::F.Invoke<A, R>::invoke::h11905936354613679413
  68:        0x10cda560f - rt::unwind::try::try_fn::h4025213131938170097
  69:        0x1108e3628 - rust_try_inner
  70:        0x1108e3615 - rust_try
  71:        0x10cda5e48 - thunk::F.Invoke<A, R>::invoke::h13725635537516470714
  72:        0x11084f0a2 - sys::thread::thread_start::h1120d6d2e105321012E
  73:     0x7fff98f8f267 - _pthread_body
  74:     0x7fff98f8f1e4 - _pthread_start
```

**Comment by XMPPwocky** [comment]
Compiling github.com/tomaka/glium/ produces the same error on win8 64-bit.
rustc 1.0.0-nightly (3b3bb0e68 2015-03-04) (built 2015-03-05)

**Comment by kubkon** [comment]
I can confirm that there is no panic with yesterday's build:

```
rustc 1.0.0-nightly (fed12499e 2015-03-03) (built 2015-03-03)
```

**Comment by XMPPwocky** [comment]
Appears fixed with glium on today's nightly.

**Comment by rprichard** [comment]
I wonder whether the overflow in the compiler was actually fixed, or whether we just stopped enabling overflow checks in the compiler.  I think `-O` recently changed so that it disables overflow checks.

**Comment by viperscape** [comment]
I received this overflow on the new rust build, I ended up changing a function from usize to accept i8, instead of using "as i8" inside the function-- which solved the issue for me

**Comment by Byron** [comment]
Seems to be related to #23407

**Comment by frewsxcv** [comment]
Does anyone know of a minimal example of this?

**Comment by arielb1** [comment]
This is the codemap span bug. It doesn't occur in new versions of rustc.
